app/src/main/java/com/celzero/bravedns/ui/NotificationHandlerDialog.kt

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2021 RethinkDNS and its authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.celzero.bravedns.ui
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.provider.Settings
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import com.celzero.bravedns.R
+import com.celzero.bravedns.service.VpnController
+import com.celzero.bravedns.util.Constants
+
+class NotificationHandlerDialog : AppCompatActivity() {
+    enum class TrampolineType {
+        ACCESSIBILITY_SERVICE_FAILURE_DIALOG,
+        NEW_APP_INSTAL_DIALOG,
+        HOMESCREEN_ACTIVITY,
+        PAUSE_ACTIVITY,
+        NONE
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        handleNotificationIntent(intent)
+    }
+
+    override fun onNewIntent(intent: Intent) {
+        super.onNewIntent(intent)
+        handleNotificationIntent(intent)
+    }
+
+    // In two-cases (accessibility failure/new app install action), the app directly launches
+    // firewall activity from notification action. Need to handle the pause state for those cases
+    private fun handleNotificationIntent(intent: Intent) {
+
+        when (determineTrampoline(intent)) {
+            TrampolineType.ACCESSIBILITY_SERVICE_FAILURE_DIALOG -> {
+                handleAccessibilitySettings()
+            }
+            TrampolineType.NEW_APP_INSTAL_DIALOG -> {
+                // navigate to all apps screen
+                launchFirewallActivity()
+            }
+            TrampolineType.HOMESCREEN_ACTIVITY -> {
+                launchHomeScreen()
+            }
+            TrampolineType.PAUSE_ACTIVITY -> {
+                showAppPauseDialog(receivedIntentType(intent))
+            }
+            TrampolineType.NONE -> {
```

Comment:
move this into its own method

```
fun trampoline(t: TrampolineType) {
    when (t) {
    
        // ...
    
    }
}

fun handleNotificationIntent(intent: Intent) {
    return trampoline(determineTrampoline(intent))
}
```

may be `determineTrampoline` can be moved in to this.

Reply:
Moved into `handleNotificationIntent`